In [1]:
# This Python 3 environment comes with many helpful analytics libraries installed
# It is defined by the kaggle/python Docker image: https://github.com/kaggle/docker-python
# For example, here's several helpful packages to load

import numpy as np # linear algebra
import pandas as pd # data processing, CSV file I/O (e.g. pd.read_csv)

# Input data files are available in the read-only "../input/" directory
# For example, running this (by clicking run or pressing Shift+Enter) will list all files under the input directory

import os
for dirname, _, filenames in os.walk('/kaggle/input'):
    for filename in filenames:
        print(os.path.join(dirname, filename))

# You can write up to 20GB to the current directory (/kaggle/working/) that gets preserved as output when you create a version using "Save & Run All" 
# You can also write temporary files to /kaggle/temp/, but they won't be saved outside of the current session

In [2]:
import pandas as pd
df = pd.read_csv('https://raw.githubusercontent.com/Datamanim/pandas/main/lol.csv', delimiter='\t')
df.head(2)

gameId   creationTime  gameDuration  seasonId  winner  firstBlood  \
0  3326086514  1504279457970          1949         9       1           2   
1  3229566029  1497848803862          1851         9       1           1   

   firstTower  firstInhibitor  firstBaron  firstDragon  ...  t2_towerKills  \
0           1               1           1            1  ...              5   
1           1               1           0            1  ...              2   

   t2_inhibitorKills  t2_baronKills  t2_dragonKills  t2_riftHeraldKills  \
0                  0              0               1                   1   
1                  0              0               0                   0   

   t2_ban1  t2_ban2  t2_ban3  t2_ban4  t2_ban5  
0      114       67       43       16       51  
1       11       67      238       51      420  

[2 rows x 61 columns]

## Question 1

데이터를 로드하라. 데이터는 \t을 기준으로 구분되어있다.

## Question 2

데이터의 상위 5개 행을 출력하라

## Question 3

데이터의 행과 열의 갯수를 파악하라

## 전체 컬럼을 출력하라

## Question 5

6번째 컬럼명을 출력하라

## Question 6

6번째 컬럼의 데이터 타입을 확인하라

## Question 7

데이터셋의 인덱스 구성은 어떤가

## Question 8

6번째 컬럼의 3번째 값은 무엇인가?


In [3]:
import pandas as pd
df = pd.read_csv('https://raw.githubusercontent.com/Datamanim/pandas/main/Jeju.csv', encoding='cp949')
df.head(2)



id          일자  시도명 읍면동명        거주인구       근무인구        방문인구      총 유동인구  \
0  22448  2018-01-01  제주시  도두동   32249.987   3418.266  102709.092  138377.345   
1  22449  2018-01-01  제주시  외도동  213500.997  10341.172  112692.789  336534.958   

    평균 속도  평균 소요 시간  평균 기온  일강수량  평균 풍속  
0  39.556    29.167    5.0   0.0    2.5  
1  32.900    30.900    5.0   0.0    2.5

## Question 9

데이터를 로드하라. 컬럼이 한글이기에 적절한 처리해줘야함


## Question 10

데이터 마지막 3개행을 출력하라

## Question 11

수치형 변수를 가진 컬럼을 출력하라

## Question 12

범주형 변수를 가진 컬럼을 출력하라

## Question 13

각 컬럼의 결측치 숫자를 파악하라

## Question 14

각 컬럼의 데이터수, 데이터타입을 한번에 확인하라


## Question 15

각 수치형 변수의 분포(사분위, 평균, 표준편차, 최대 , 최소)를 확인하라


## Question 16

거주인구 컬럼의 값들을 출력하라

## Question 17

평균 속도 컬럼의 4분위 범위(IQR) 값을 구하여라

## Question 18

읍면동명 컬럼의 유일값 갯수를 출력하라

## Question 19

읍면동명 컬럼의 유일값을 모두 출력하라


In [4]:
import pandas as pd
df = pd.read_csv('https://raw.githubusercontent.com/Datamanim/pandas/main/chipo.csv')
df.head(2)

order_id  quantity                     item_name choice_description  \
0         1         1  Chips and Fresh Tomato Salsa                NaN   
1         1         1                          Izze       [Clementine]   

  item_price  
0     $2.39   
1     $3.39


# 02 Filtering & Sorting

## Question 20

데이터를 로드하라.

## Question 21

quantity컬럼 값이 3인 데이터를 추출하여 첫 5행을 출력하라

## Question 22

quantity컬럼 값이 3인 데이터를 추출하여 index를 0부터 정렬하고 첫 5행을 출력하라

## Question 23

quantity , item_price 두개의 컬럼으로 구성된 새로운 데이터 프레임을 정의하라


## Question 24

item_price 컬럼의 달러표시 문자를 제거하고 float 타입으로 저장하여 new_price 컬럼에 저장하라

## Question 25

new_price 컬럼이 5이하의 값을 가지는 데이터프레임을 추출하고, 전체 갯수를 구하여라

## Question 26

item_name명이 Chicken Salad Bowl 인 데이터 프레임을 추출하라고 index 값을 초기화 하여라

## Question 27

new_price값이 9 이하이고 item_name 값이 Chicken Salad Bowl 인 데이터 프레임을 추출하라

## Question 28

df의 new_price 컬럼 값에 따라 오름차순으로 정리하고 index를 초기화 하여라

## Question 29

df의 item_name 컬럼 값중 Chips 포함하는 경우의 데이터를 출력하라

## Question 30

df의 짝수번째 컬럼만을 포함하는 데이터프레임을 출력하라

## Question 31

df의 new_price 컬럼 값에 따라 내림차순으로 정리하고 index를 초기화 하여라



## Question 32

df의 item_name 컬럼 값이 Steak Salad 또는 Bowl 인 데이터를 인덱싱하라

## Question 33

df의 item_name 컬럼 값이 Steak Salad 또는 Bowl 인 데이터를 데이터 프레임화 한 후, item_name를 기준으로 중복행이 있으면 제거하되 첫번째 케이스만 남겨라


## Question 34

df의 item_name 컬럼 값이 Steak Salad 또는 Bowl 인 데이터를 데이터 프레임화 한 후, item_name를 기준으로 중복행이 있으면 제거하되 마지막 케이스만 남겨라



## Question 35

df의 데이터 중 new_price값이 new_price값의 평균값 이상을 가지는 데이터들을 인덱싱하라


## Question 36

df의 데이터 중 item_name의 값이 Izze 데이터를 Fizzy Lizzy로 수정하라

## Question 37

df의 데이터 중 choice_description 값이 NaN 인 데이터의 갯수를 구하여라



## Question 38

df의 데이터 중 choice_description 값이 NaN 인 데이터를 NoData 값으로 대체하라(loc 이용)


## Question 39

df의 데이터 중 choice_description 값에 Black이 들어가는 경우를 인덱싱하라

## Question 40

df의 데이터 중 choice_description 값에 Vegetables 들어가지 않는 경우의 갯수를 출력하라

## Question 41

df의 데이터 중 item_name 값이 N으로 시작하는 데이터를 모두 추출하라

## Question 42

df의 데이터 중 item_name 값의 단어갯수가 15개 이상인 데이터를 인덱싱하라

## Question 43

df의 데이터 중 new_price값이 lst에 해당하는 경우의 데이터 프레임을 구하고 그 갯수를 출력하라 lst =[1.69, 2.39, 3.39, 4.45, 9.25, 10.98, 11.75, 16.98]


In [5]:
import pandas as pd
df= pd.read_csv('https://raw.githubusercontent.com/Datamanim/pandas/main/AB_NYC_2019.csv')
df.head(2)

id                                name  host_id host_name  \
0  2539  Clean & quiet apt home by the park     2787      John   
1  2595               Skylit Midtown Castle     2845  Jennifer   

  neighbourhood_group neighbourhood  latitude  longitude        room_type  \
0            Brooklyn    Kensington  40.64749  -73.97237     Private room   
1           Manhattan       Midtown  40.75362  -73.98377  Entire home/apt   

   price  minimum_nights  number_of_reviews last_review  reviews_per_month  \
0    149               1                  9  2018-10-19               0.21   
1    225               1                 45  2019-05-21               0.38   

   calculated_host_listings_count  availability_365  
0                               6               365  
1                               2               355

# 03_Grouping

## Question 44

데이터를 로드하고 상위 5개 컬럼을 출력하라

## Question 45

데이터의 각 host_name의 빈도수를 구하고 host_name으로 정렬하여 상위 5개를 출력하라

## Question 46

데이터의 각 host_name의 빈도수를 구하고 빈도수 기준 내림차순 정렬한 데이터 프레임을 만들어라. 빈도수 컬럼은 counts로 명명하라

## Question 47

neighbourhood_group의 값에 따른 neighbourhood컬럼 값의 갯수를 구하여라

## Question 48

neighbourhood_group의 값에 따른 neighbourhood컬럼 값 중 neighbourhood_group그룹의 최댓값들을 출력하라

## Question 49

neighbourhood_group 값에 따른 price값의 평균, 분산, 최대, 최소 값을 구하여라

## Question 50

neighbourhood_group 값에 따른 reviews_per_month 평균, 분산, 최대, 최소 값을 구하여라

## Question 51

neighbourhood 값과 neighbourhood_group 값에 따른 price 의 평균을 구하라

## Question 52

neighbourhood 값과 neighbourhood_group 값에 따른 price 의 평균을 계층적 indexing 없이 구하라

## Question 53

neighbourhood 값과 neighbourhood_group 값에 따른 price 의 평균을 계층적 indexing 없이 구하고 nan 값은 -999값으로 채워라

## Question 54

데이터중 neighbourhood_group 값이 Queens값을 가지는 데이터들 중 neighbourhood 그룹별로 price값의 평균, 분산, 최대, 최소값을 구하라

## Question 55

데이터중 neighbourhood_group 값에 따른 room_type 컬럼의 숫자를 구하고 neighbourhood_group 값을 기준으로 각 값의 비율을 구하여라

In [6]:
import pandas as pd
df = pd.read_csv('https://raw.githubusercontent.com/Datamanim/pandas/main/BankChurnersUp.csv')
df.head(2)

Unnamed: 0  CLIENTNUM     Attrition_Flag  Customer_Age Gender  \
0           0  768805383  Existing Customer            45      M   
1           1  818770008  Existing Customer            49      F   

   Dependent_count Education_Level Marital_Status Income_Category  \
0                3     High School        Married     $60K - $80K   
1                5        Graduate         Single  Less than $40K   

  Card_Category  Months_on_book  Total_Relationship_Count  \
0          Blue              39                         5   
1          Blue              44                         6   

   Months_Inactive_12_mon  Contacts_Count_12_mon  Credit_Limit  \
0                       1                      3       12691.0   
1                       1                      2        8256.0   

   Total_Revolving_Bal  Avg_Open_To_Buy  Total_Amt_Chng_Q4_Q1  Total_Trans_Amt  
0                  777          11914.0                 1.335             1144  
1                  864           7392.0                 1.541             1291

# 04_Apply , Map

## Question 56

데이터를 로드하고 데이터 행과 열의 갯수를 출력하라

## Question 57

Income_Category의 카테고리를 map 함수를 이용하여 다음과 같이 변경하여 newIncome 컬럼에 매핑하라 Unknown : N
Less than $40K : a
$40K - $60K : b
$60K - $80K : c
$80K - $120K : d
$120K +’ : e

## Question 58

Income_Category의 카테고리를 apply 함수를 이용하여 다음과 같이 변경하여 newIncome 컬럼에 매핑하라 Unknown : N
Less than $40K : a
$40K - $60K : b
$60K - $80K : c
$80K - $120K : d
$120K +’ : e

## Question 59

Customer_Age의 값을 이용하여 나이 구간을 AgeState 컬럼으로 정의하라. (0~9 : 0 , 10~19 :10 , 20~29 :20 … 각 구간의 빈도수를 출력하라

## Question 60

Education_Level의 값중 Graduate단어가 포함되는 값은 1 그렇지 않은 경우에는 0으로 변경하여 newEduLevel 컬럼을 정의하고 빈도수를 출력하라

## Question 61

Credit_Limit 컬럼값이 4500 이상인 경우 1 그외의 경우에는 모두 0으로 하는 newLimit 정의하라. newLimit 각 값들의 빈도수를 출력하라

## Question 62

Marital_Status 컬럼값이 Married 이고 Card_Category 컬럼의 값이 Platinum인 경우 1 그외의 경우에는 모두 0으로 하는 newState컬럼을 정의하라. newState의 각 값들의 빈도수를 출력하라

## Question 63

Gender 컬럼값 M인 경우 male F인 경우 female로 값을 변경하여 Gender 컬럼에 새롭게 정의하라. 각 value의 빈도를 출력하라

In [7]:
import pandas as pd
df = pd.read_csv('https://raw.githubusercontent.com/Datamanim/pandas/main/timeTest.csv')
df.head(2)

Yr_Mo_Dy    RPT    VAL    ROS   KIL    SHA   BIR    DUB    CLA    MUL  \
0  2061-01-01  15.04  14.96  13.17  9.29    NaN  9.87  13.67  10.25  10.83   
1  2061-01-02  14.71    NaN  10.83  6.50  12.62  7.67  11.50  10.04   9.79   

     CLO    BEL    MAL  
0  12.58  18.50  15.04  
1   9.67  17.54  13.83


## Question 64

데이터를 로드하고 각 열의 데이터 타입을 파악하라

## Question 65

Yr_Mo_Dy을 판다스에서 인식할 수 있는 datetime64타입으로 변경하라

## Question 66

Yr_Mo_Dy에 존재하는 년도의 유일값을 모두 출력하라

## Question 67

Yr_Mo_Dy에 년도가 2061년 이상의 경우에는 모두 잘못된 데이터이다. 해당경우의 값은 100을 빼서 새롭게 날짜를 Yr_Mo_Dy 컬럼에 정의하라

## Question 68

년도별 각컬럼의 평균값을 구하여라

## Question 69

weekday컬럼을 만들고 요일별로 매핑하라 ( 월요일: 0 ~ 일요일 :6)

## Question 70

weekday컬럼을 기준으로 주말이면 1 평일이면 0의 값을 가지는 WeekCheck 컬럼을 만들어라

## Question 71

년도, 일자 상관없이 모든 컬럼의 각 달의 평균을 구하여라

## Question 72

모든 결측치는 컬럼기준 직전의 값으로 대체하고 첫번째 행에 결측치가 있을경우 뒤에있는 값으로 대채하라

## Question 73

년도 - 월을 기준으로 모든 컬럼의 평균값을 구하여라

## Question 74

RPT 컬럼의 값을 일자별 기준으로 1차차분하라

## Question 75

RPT와 VAL의 컬럼을 일주일 간격으로 각각 이동평균한값을 구하여라

In [8]:
import pandas as pd
df =pd.read_csv('https://raw.githubusercontent.com/Datamanim/pandas/main/seoul_pm.csv')
df.head(2)

(년-월-일:시) PM10등급  PM10 PM2.5등급  PM2.5 오존등급     오존 이산화질소등급  이산화질소  \
0  2021-05-15:15     보통  47.0      보통   19.0   좋음  0.017      좋음  0.023   
1  2021-05-15:14     보통  43.0      보통   20.0   좋음  0.024      좋음  0.019   

  일산화탄소등급  일산화탄소 아황산가스등급  아황산가스  
0      좋음    0.4      좋음  0.003  
1      좋음    0.3      좋음  0.003


## Question 76

년-월-일:시 컬럼을 pandas에서 인식할 수 있는 datetime 형태로 변경하라. 서울시의 제공데이터의 경우 0시가 24시로 표현된다

## Question 77

일자별 영어요일 이름을 dayName 컬럼에 저장하라

## Question 78

일자별 각 PM10등급의 빈도수를 파악하라

## Question 79

시간이 연속적으로 존재하며 결측치가 없는지 확인하라

## Question 80

오전 10시와 오후 10시(22시)의 PM10의 평균값을 각각 구하여라

## Question 81

날짜 컬럼을 index로 만들어라

## Question 82

데이터를 주단위로 뽑아서 최소,최대 평균, 표준표차를 구하여라

In [9]:
import pandas as pd
df = pd.read_csv('https://raw.githubusercontent.com/Datamanim/pandas/main/under5MortalityRate.csv')
df.head(2)

Location  Period                                          Indicator  \
0  Afghanistan    2019  Under-five mortality rate (probability of dyin...   
1  Afghanistan    2019  Under-five mortality rate (probability of dyin...   

         Dim1        First Tooltip  
0  Both sexes  60.27 [47.44-74.62]  
1        Male  63.83 [50.04-79.16]


## Question 83

Indicator을 삭제하고 First Tooltip 컬럼에서 신뢰구간에 해당하는 표현을 지워라

## Question 84

년도가 2015년 이상, Dim1이 Both sexes인 케이스만 추출하라

## Question 85

84번 문제에서 추출한 데이터로 아래와 같이 나라에 따른 년도별 사망률을 데이터 프레임화 하라

## Question 86

Dim1에 따른 년도별 사망비율의 평균을 구하라

In [10]:
import pandas as pd
df = pd.read_csv('https://raw.githubusercontent.com/Datamanim/pandas/main/winter.csv')
df.head(2)

Year      City     Sport Discipline         Athlete Country Gender  \
0  1924  Chamonix  Biathlon   Biathlon     BERTHET, G.     FRA    Men   
1  1924  Chamonix  Biathlon   Biathlon  MANDRILLON, C.     FRA    Men   

             Event   Medal  
0  Military Patrol  Bronze  
1  Military Patrol  Bronze


## Question 87

데이터에서 한국 KOR 데이터만 추출하라

## Question 88

한국 올림픽 메달리스트 데이터에서 년도에 따른 medal 갯수를 데이터프레임화 하라

## Question 89

전체 데이터에서 sport종류에 따른 성별수를 구하여라

## Question 90

전체 데이터에서 Discipline종류에 따른 따른 Medal수를 구하여라

In [11]:
import pandas as pd
df = pd.read_csv('https://raw.githubusercontent.com/Datamanim/pandas/main/mergeTEst.csv',index_col= 0)
df.head(2)

df1 = df.iloc[:4,:]
df2 = df.iloc[4:,:]

df3 = df.iloc[:2,:4]
df4 = df.iloc[5:,3:]

df5 = df.T.iloc[:7,:3]
df6 = df.T.iloc[6:,2:5]

display(df1)
display(df2)

2010    2011    2012    2013   2014    2015    2016    2017  \
Location                                                                     
Afghanistan  64.023  61.640  59.367  57.170  55.08  53.107  51.267  49.560   
Albania      11.803  10.807   9.943   9.267   8.79   8.493   8.363   8.363   
Algeria      23.540  22.907  22.450  22.117  21.85  21.587  21.257  20.850   
Andorra       4.240   4.033   3.843   3.667   3.49   3.330   3.187   3.060   

               2018    2019  
Location                     
Afghanistan  47.983  46.453  
Albania       8.453   8.597  
Algeria      20.407  19.930  
Andorra       2.933   2.827

2010    2011    2012    2013    2014    2015    2016  \
Location                                                                      
Angola               75.713  71.280  67.233  63.570  60.430  57.757  55.510   
Antigua and Barbuda   8.667   8.223   7.807   7.420   7.070   6.757   6.483   
Argentina            12.887  12.380  11.840  11.283  10.733  10.203   9.683   
Armenia              16.497  15.677  14.897  14.170  13.477  12.817  12.183   
Australia             3.993   3.803   3.623   3.467   3.343   3.253   3.183   
Austria               3.573   3.463   3.333   3.210   3.113   3.043   2.987   

                       2017    2018    2019  
Location                                     
Angola               53.460  51.757  50.093  
Antigua and Barbuda   6.230   6.000   5.783  
Argentina             9.177   8.680   8.227  
Armenia              11.583  11.007  10.497  
Australia             3.137   3.090   3.047  
Austria               2.943   2.897   2.843

# 07_Merge , Concat

## Question 91

df1과 df2 데이터를 하나의 데이터 프레임으로 합쳐라

## Question 92

df3과 df4 데이터를 하나의 데이터 프레임으로 합쳐라. 둘다 포함하고 있는 년도에 대해서만 고려한다


## Question 93

df3과 df4 데이터를 하나의 데이터 프레임으로 합쳐라. 모든 컬럼을 포함하고, 결측치는 0으로 대체한다

## Question 94

df5과 df6 데이터를 하나의 데이터 프레임으로 merge함수를 이용하여 합쳐라. Algeria컬럼을 key로 하고 두 데이터 모두 포함하는 데이터만 출력하라

## Question 95

df5과 df6 데이터를 하나의 데이터 프레임으로 merge함수를 이용하여 합쳐라. Algeria컬럼을 key로 하고 합집합으로 합쳐라